pkg/endpoint/endpoint.go

```diff
@@ -77,32 +77,35 @@ var (
 	EndpointMutableOptionLibrary = option.GetEndpointMutableOptionLibrary()
 )
 
+// State is an enumeration for possible endpoint states.
+type State string
```

Comment:
Out of curiosity, why not `type State models.EndpointState`?

Reply:
That would work too.
The choice here is more a matter of style and showing an intention.

That was my way of thinking about that:

Just using string as a base is simple and inspecting that code is easy. 
By seeing string as the underlying type, we don't have to dig deeper.

Although we are reusing its values, `model.EndpointState` comes from generated code, so it's not really under our control.
Avoiding coupling with it where possible seems a good idea. 

Making State an independent type is more clear when the enum has to be extended with a value that doesn't have equivalent in the model.
E.g. if I'd like to add "UndefinedState" to name the zero value of `endpoint.state` (to replace `""` used here and there), that value wouldn't have an explicit counterpart in the model, but type derived from `model.EndpointState` may suggest otherwise.